src/EventSubscriber/LinkTypePlusEventSubscriber.php

```diff
@@ -63,7 +64,110 @@ public function onlinkTypeCreditCardPaymentEvent(LinkTypePlusEvent $event) {
    */
   public function onlinkTypePayPayEvent(LinkTypePlusEvent $event) {
     $paymentMethod = $event->getPaymentMethod();
-    $this->loggerFactory->notice("$paymentMethod payment event has been subscribed");
-    return TRUE;
+   //Get the payment status and update the payment in Drupal
+   $orderId = $event->getOrderId();
+   $status = $event->getTransitionState();
+   $drupalStatus = $this->webhookStatusMapper($status);
+   $remoteId = $event->getRemoteId();
+   $this->loggerFactory->notice("$paymentMethod payment event has been subscribed");
+   $this->loggerFactory->notice("$orderId \n  $status \n $remoteId");
+   if($this->updatePaymentStatus($orderId, $drupalStatus, $remoteId)){
+     $this->loggerFactory->notice('Status has been updated');
+     return TRUE;
+   }
+  }
+
+
+  /**
+   * Update the status in Drupal.
+   *
+   * @param array $data
+   *   The GMO api response data.
+   */
+  public function updatePaymentStatus($order_id, $status, $remote_id){
```

Comment:
In eventSubscriber we can return the drupal status to controller where we can perform the updatePaymentStatus

This UpdatePaymentStatus not going to alter whatever the payment method we are using